### PR TITLE
[W5.7][W12-4]Spencer Tan

### PR DESCRIPTION
**Pull request by spencertan96**
Added Notes feature, a new detail about persons in the AddressBook that the user can add
Updated existing tests and added a few new tests.
Updated user guide and developer guide's class diagram.

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -50,22 +50,22 @@ Help is also shown if you enter an incorrect command e.g. `abcd`
 == Adding a person: `add`
 
 Adds a person to the address book. +
-Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
+Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [p]n/NOTES [t/TAG]...`
```
Comment (Zackkh95) [inline docs/UserGuide.adoc:53]: Format: add in notes

File: src/seedu/addressbook/data/person/Notes.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package seedu.addressbook.data.person;
```
Comment (Zackkh95) [inline src/seedu/addressbook/data/person/Notes.java:1]: Creates notes class

**Review by Zackkh95 — COMMENTED** [review]
Good job Spencer!

**Review by Zackkh95 — APPROVED** [review]
Good job Spencer!

File: src/seedu/addressbook/data/person/Notes.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Represents notes about a Person in the address book.
+ */
+public class Notes {
+
+    public static final String EXAMPLE = "Hardworking student.";
+    public String value;
+    private boolean isPrivate;
+
+    /**
+     *  Initialises value
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Notes.java:13]: Our coding standard requires to use American spelling.

File: src/seedu/addressbook/parser/Parser.java [hunk 4]
```diff
@@ -39,11 +39,13 @@
...
                     + " (?<isEmailPrivate>p?)e/(?<email>[^/]+)"
                     + " (?<isAddressPrivate>p?)a/(?<address>[^/]+)"
+                    + " (?<isNotesPrivate>p?)n/(?<notes>[^/]+)"
                     + "(?<tagArguments>(?: t/[^/]+)*)"); // variable number of tags
 
 
     /**
-     * Signals that the user input could not be parsed.
+     * Signals that the user input could not be parsed:w
+     * .
```
Comment (damithc) [inline src/seedu/addressbook/parser/Parser.java:48]: Unintended change? It is a good practice to check the diff of the PR and fix any problems before submitting it.

File: test/java/seedu/addressbook/util/TestUtil.java [hunk 5]
```diff
@@ -16,12 +16,7 @@
 
 import seedu.addressbook.data.AddressBook;
 import seedu.addressbook.data.exception.IllegalValueException;
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.*;
```
Comment (damithc) [inline test/java/seedu/addressbook/util/TestUtil.java:19]: [Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation:exclamation:

**Review by damithc — APPROVED** [review]
:+1: Good effort on this PR @spencertan96. In particular, good job on,
* updating docs
* keeping the PR relatively clean

Suggestions to improve further:
* update I/O tests to test the new functionality
* update JUnit tests to test the new functionality
* pay closer attention to the coding standard

As before, please close the PR after viewing comments.

**Pull request closed without merging**